plugins/package-managers/node/src/main/kotlin/npm/Npm.kt

```diff
@@ -38,10 +38,10 @@ import org.ossreviewtoolkit.plugins.packagemanagers.node.NodePackageManagerType
 import org.ossreviewtoolkit.plugins.packagemanagers.node.PackageJson
 import org.ossreviewtoolkit.plugins.packagemanagers.node.parsePackageJson
 import org.ossreviewtoolkit.utils.common.CommandLineTool
+import org.ossreviewtoolkit.utils.common.DirectoryStash
```

Comment:
I still have to think about this commit a bit, haven't finished the review.

The previous commits LGTM now. Do you want to split out this commit so you can merge the rest ?

Reply:
> Do you want to split out this commit so you can merge the rest ?

I'm not sure... since this is (to me) the most important commit that concludes the series and allows for the React-Native NPM / Gradle use-case to be analyzed, there is not much value in merging the other commits without this.

> I still have to think about this commit a bit

Maybe it helps if you look at the current implementation of `stashDirectories()`: It's basically just a shortcut for creating a `DirectoryStash` instance, that is then closed via `use {}`. All that the last commit intends to do is to split the creation of the `DirectoryStash` instance and the closing across `beforeResolution()` / `afterResolution()` by calling `close()` explicitly instead of leveraging `use {}`.

Reply:
> Maybe it helps if you look at the current implementation of `stashDirectories()`

I got that bit, but I need to first better understand some details of the analyzer, e.g. how this fits with mapDefinitionFiles, but also in general.

Reply:
Could we also address issues, if any, as a follow-up? I'd like to include this feature into tomorrow's ORT release.

Regarding the mapping of definition files: That happens in `findManagedFiles()` before `analyze{InParallel}()` is called. So `{before,after}Resolution()` see the mapped files, just like originally, if that was one of the questions.

Reply:
> The previous commits LGTM now. Do you want to split out this commit so you can merge the rest ?

Despite my original plan, I've done that now to move things forward. Please approve @fviernau: https://github.com/oss-review-toolkit/ort/pull/9824

Reply:
> In general, I believe it'd be nice to also have @mnonnenmacher's view.

Looks good to me, I don't see any issues with the approach.